components/channel_notifications_modal/index.js

```diff
@@ -0,0 +1,26 @@
+import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+
+import ChannelNotificationsModal from './channel_notifications_modal.jsx';
+
+import {NotificationLevels} from 'utils/constants.jsx';
+
+import {updateChannelNotifyProps} from 'actions/channel_actions.jsx';
+
+function mapStateToProps(state, ownProps) {
+    return {
+        ...ownProps,
+        notifyLevel: ownProps.channelMember.notify_props.desktop,
+        unreadLevel: ownProps.channelMember.notify_props.mark_unread,
+        pushLevel: ownProps.channelMember.notify_props.push || NotificationLevels.DEFAULT
+    };
```

Comment:
I feel like above props are not necessary to define here since those are from `props.channelMember` already.